tests/dialects/test_tsql.py

```diff
@@ -2009,3 +2009,57 @@ def test_grant(self):
         self.validate_identity(
             "GRANT EXECUTE ON TestProc TO User2 AS TesterRole", check_command_warning=True
         )
+
+    def test_parsename(self):
+        # Test default case
+        self.validate_all(
+            "SELECT PARSENAME('1.2.3', 1)",
+            read={
+                "spark": "SELECT SPLIT_PART('1.2.3', '.', 3)",
+                "databricks": "SELECT SPLIT_PART('1.2.3', '.', 3)",
+            },
+            write={
+                "spark": "SELECT SPLIT_PART('1.2.3', '.', 3)",
+                "databricks": "SELECT SPLIT_PART('1.2.3', '.', 3)",
+                "tsql": "SELECT PARSENAME('1.2.3', 1)",
+            },
+        )
+        # Test zero index
+        self.validate_all(
+            "SELECT PARSENAME('1.2.3', 0)",
+            read={
+                "spark": "SELECT SPLIT_PART('1.2.3', '.', 4)",
+                "databricks": "SELECT SPLIT_PART('1.2.3', '.', 4)",
+            },
+            write={
+                "spark": "SELECT SPLIT_PART('1.2.3', '.', 4)",
+                "databricks": "SELECT SPLIT_PART('1.2.3', '.', 4)",
+                "tsql": "SELECT PARSENAME('1.2.3', 0)",
+            },
+        )
+        # Test null value
+        self.validate_all(
+            "SELECT PARSENAME(NULL, 1)",
+            read={
+                "spark": "SELECT SPLIT_PART(NULL, '.', 1)",
+                "databricks": "SELECT SPLIT_PART(NULL, '.', 1)",
+            },
+            write={
+                "spark": "SELECT SPLIT_PART(NULL, '.', 1)",
+                "databricks": "SELECT SPLIT_PART(NULL, '.', 1)",
+                "tsql": "SELECT PARSENAME(NULL, 1)",
+            },
+        )
+        # Test non-dot delimiter
+        self.validate_all(
+            "SELECT SPLIT_PART('1.2.3', ',', 1)",
+            read={
+                "spark": "SELECT SPLIT_PART('1.2.3', ',', 1)",
+                "databricks": "SELECT SPLIT_PART('1.2.3', ',', 1)",
+            },
+            write={
+                "spark": "SELECT SPLIT_PART('1.2.3', ',', 1)",
+                "databricks": "SELECT SPLIT_PART('1.2.3', ',', 1)",
+                "tsql": "SELECT SPLIT_PART('1.2.3', ',', 1)",
```

Comment:
This should yield `UnsupportedError` if the generator code is changed to `self.unsupported(...)`, it's more robust than generating a non-existing function